### PR TITLE
Improve qualifications section of the application details page

### PR DESCRIPTION
**Pull request by simonwhatley**
See Trello cards:

- [Improving the qualifications section of the application details page](https://trello.com/c/tkB4TwLX)
- [Showing information about English GCSE when they do not have an English GCSE or equivalent](https://trello.com/c/XfJy7wMJ)

### CODE REVIEW

File: app/data/generators/degree.js [hunk 1]
```diff
@@ -1,67 +1,86 @@
...
-const degreeData = require('../degree')
+const DegreeHelper = require('../helpers/degrees')
 
-module.exports = (isInternationCandidate) => {
-  const item = () => {
-    const subject = faker.helpers.randomize(degreeData().subjects)
-    const predicted = faker.datatype.boolean()
-    const startDate = '2017'
-    const endDate = '2020'
+module.exports = (isInternationCandidate, dateOfBirth) => {
```
Comment (adamsilver) [inline app/data/generators/degree.js:8]: Change to an object { isInternationalCandidate, dateOfBirth}.

You also have a typo.

File: app/data/generators/degree.js [hunk 2]
```diff
@@ -1,67 +1,86 @@
...
+
+  // Start year - 18/19 years after date of birth
+  let year = DateTime.fromISO(dateOfBirth).toObject().year
```
Comment (adamsilver) [inline app/data/generators/degree.js:11]: Is that 18 OR 19 years?

File: app/data/generators/degree.js [hunk 3]
```diff
@@ -1,67 +1,86 @@
...
+      degree.startYear = year
+    }
+
+    // Set the graduation year to be between 3 and 4 years after start year
+    degree.graduationYear = degree.startYear + faker.datatype.number({ 'min': 3, 'max': 4 })
+
+    // If graduation year is after the current year, set predicted to true
+    degree.predicted = (degree.graduationYear > currentYear) ? true : false
 
     if (isInternationCandidate) {
```
Comment (adamsilver) [inline app/data/generators/degree.js:42]: Typoe

File: app/data/generators/english-language-qualification.js [hunk 4]
```diff
@@ -1,18 +1,39 @@
 const faker = require('faker')
 faker.locale = 'en_GB'
 
-module.exports = () => {
+const { DateTime } = require('luxon')
+const weighted = require('weighted')
+
+module.exports = (englishGcseQualification, dateOfBirth) => {
```
Comment (adamsilver) [inline app/data/generators/english-language-qualification.js:7]: Make object as above.

File: app/filters.js [hunk 5]
```diff
@@ -179,6 +179,17 @@ module.exports = (env) => {
     else return numbers[number -1]
   }
 
+  filters.ordinal = (number) => {
+    const ordinals = [ 'first', 'second', 'third', 'fourth', 'fifth', 'sixth', 'seventh', 'eighth', 'ninth' ];
```
Comment (adamsilver) [inline app/filters.js:183]: What happens for tenth?

**Review by adamsilver — CHANGES_REQUESTED** [review]
Left comments inline.

File: app/data/generators/degree.js (hunk 2 above)
Comment (simonwhatley) [reply app/data/generators/degree.js:11]: Yes. I'll fix.

File: app/filters.js (hunk 5 above)
Comment (simonwhatley) [reply app/filters.js:183]: We spell out first to ninth. After that, we use 10th, 11th and so on.

In tables, we should use numerals throughout.

See GOV.UK style guide: https://www.gov.uk/guidance/style-guide/a-to-z-of-gov-uk-style#ordinal-numbers

File: app/data/generators/degree.js (hunk 1 above)
Comment (simonwhatley) [reply app/data/generators/degree.js]: Fixed

File: app/data/generators/degree.js (hunk 2 above)
Comment (simonwhatley) [reply app/data/generators/degree.js]: Comment updated to say "18 or 19"

File: app/data/generators/degree.js (hunk 3 above)
Comment (simonwhatley) [reply app/data/generators/degree.js]: Fixed

File: app/data/generators/english-language-qualification.js (hunk 4 above)
Comment (simonwhatley) [reply app/data/generators/english-language-qualification.js]: Fixed

File: app/filters.js (hunk 5 above)
Comment (simonwhatley) [reply app/filters.js:191]: Comment added to the filter code

**Pull request merged**